src/components/PhotoGallery/index.js

```diff
@@ -1,123 +1,117 @@
+import axios from 'axios'
+
+const FLICKR_API_KEY = '381c0c551c89c0f23e326456eae0c6a8'
+const FLICKR_PHOTOSET_ID = '72157708626862634'
+const FLICKR_USER_ID = '51450332@N02'
+const FLICKR_ARMADA_URL = 'https://www.flickr.com/photos/armadakth/'
+
+const Photo = props => {
+    const fileURL = props.fileURL
+    const pageURL = props.pageURL
+    const bgStyle = {
+        backgroundImage: `url('${fileURL}')`,
+        backgroundSize: 'cover',
+        backgroundPosition: 'center',
     }
+    return (
+        <Zoom>
+            <div className='photo'>
+                <div className='square' style={bgStyle}>
+                    <a
+                        className='photo-info'
+                        target='_blank'
+                        rel='noreferrer'
+                        href={pageURL}
+                    >
+                        {props.title}
+                    </a>
+                </div>
+            </div>
+        </Zoom>
+    )
 }
 
 Photo.propTypes = {
     fileURL: PropTypes.string,
     pageURL: PropTypes.string,
     title: PropTypes.string,
-};
-
-class PhotoGallery extends React.Component {
-    constructor(props) {
-        super(props);
-
-        this.state = {
-            loaded: false,
-            photos: [],
-            startFrom: 0
-        };
-    }
-
-    componentDidMount() {
-        fetch(this.generateFlickrApiURL())
-            .then(res => res.json())
-            .then(res => {
-                this.setState({
-                    loaded: true,
-                    photos: res.photoset.photo
-                });
-                this.interval = setInterval(() => this.cyclePhotos(), 10000);
-            })
-            .catch(err => {
-                this.setState({
-                    error: err
-                });
-            });
-    }
+}
 
-    componentWillUnmount() {
-        clearInterval(this.interval);
-      }
+const PhotoGallery = props => {
+    const [loaded, setLoaded] = useState(false)
+    const [photos, setPhotos] = useState([])
+    const [startFrom, next] = useReducer(
+        state => (state + props.photoCount) % photos.length,
+        0
+    )
+
+    useEffect(() => {
+        const effect = async () => {
+            const res = await axios.get(
+                'https://api.flickr.com/services/rest/',
+                {
+                    params: {
+                        method: 'flickr.photosets.getPhotos',
+                        api_key: FLICKR_API_KEY,
+                        photoset_id: FLICKR_PHOTOSET_ID,
+                        user_id: FLICKR_USER_ID,
+                        format: 'json',
+                        nojsoncallback: '1',
+                    },
+                }
+            )
+            setLoaded(true)
+            setPhotos(res.data.photoset.photo)
+            const interval = setInterval(() => next(), 10000)
+            return () => clearInterval(interval)
+        }
+        effect()
+    }, [])
```

Comment:
```suggestion
    useEffect(() => {
        const effect = async () => {
            const res = await axios.get(
                'https://api.flickr.com/services/rest/',
                {
                    params: {
                        method: 'flickr.photosets.getPhotos',
                        api_key: FLICKR_API_KEY,
                        photoset_id: FLICKR_PHOTOSET_ID,
                        user_id: FLICKR_USER_ID,
                        format: 'json',
                        nojsoncallback: '1',
                    },
                }
            )
            setLoaded(true)
            setPhotos(res.data.photoset.photo)
        }
        effect()
    }, [])

    useEffect(() => {
        const interval = setInterval(() => next(), 10000)
        return () => clearInterval(interval)
    }, [])
```

The interval should be called separate to avoid doing multiple requests to the flickr api.

Reply:
I don't really understand this change. Can you clarify?

Reply:
I'm fairly certain that this only makes one request in all cases. I'm not sure I understand this change? Doesn't this introduce a race condition (interval before images are loaded)?

Reply:
I don't think so, because the images are reloading for me every time they switch when I run your version. With my version they don't.

Reply:
Bit it looks pretty nice though. I might do that.